dptb/nn/energy.py

```
@@ -163,47 +163,60 @@ def solve(self, h_container, s_container, kpoints:  Union[list, torch.Tensor, np
+                solver.factorize(A)
+                
+                def matvec(b):
+                    return solver.solve(A, b)
+                    
+                Op = LinearOperator((N, N), matvec=matvec, dtype=A.dtype)
+                
+                try:
+                    # Use larger NCV to help convergence, especially for clustered eigenvalues
+                    ncv =  max(2*self.neig + 1, 20)
+                    vals, vecs = eigsh(A=hk, M=M, k=self.neig, sigma=0.0, OPinv=Op, mode=self.mode, which="LM", ncv=ncv)
+                except Exception:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)